code/modules/reagents/chemistry/machinery/chem_dispenser.dm

```diff
@@ -65,6 +72,13 @@
 	if(energy != oldenergy)
 		use_power(2500)
 
+/obj/machinery/chem_dispenser/emag_act(mob/user)
+	if(emagged || !emagged_reagents)
+		user << "<span class='warning'>\The [src] has no functional safeties to emag.</span>"
+		return
+	user << "<span class='notice'>You short out \the [src]'s safeties.</span>"
+	dispensable_reagents |= emagged_reagents//add the emagged reagents to the dispensable ones
```

Comment:
You should not need to check that the `emagged_reagents` list has elements in it, because you have no code that modifies it. If you are planning to allow the list to change or be empty during gameplay, then checking that it has entries via `emagged_reagents.len`. If it has a length greater than 0, it has entries.